CHANGELOG.md

```diff
@@ -1,32 +1,120 @@
 # Changelog
 
 ## Cargo 1.84 (2025-01-09)
-[15fbd2f6...HEAD](https://github.com/rust-lang/cargo/compare/15fbd2f6...HEAD)
+[15fbd2f6...rust-1.84.0](https://github.com/rust-lang/cargo/compare/15fbd2f6...rust-1.84.0)
 
 ### Added
 
+- 🎉 Stabilize resolver v3, a.k.a the MSRV-aware dependency resolver.
+  The stabilization includes `package.resolver = "3"` in Cargo.toml,
+  and the `[resolver]` table in Cargo configuration.
+  ([RFC 3537](https://github.com/rust-lang/rfcs/blob/master/text/3537-msrv-resolver.md))
+  ([manifest docs](https://doc.rust-lang.org/nightly/cargo/reference/resolver.html#resolver-versions))
+  ([config docs](https://doc.rust-lang.org/nightly/cargo/reference/config.html#resolver))
+  [#14639](https://github.com/rust-lang/cargo/pull/14639)
+  [#14662](https://github.com/rust-lang/cargo/pull/14662)
+  [#14711](https://github.com/rust-lang/cargo/pull/14711)
+  [#14725](https://github.com/rust-lang/cargo/pull/14725)
+  [#14748](https://github.com/rust-lang/cargo/pull/14748)
+  [#14753](https://github.com/rust-lang/cargo/pull/14753)
+  [#14754](https://github.com/rust-lang/cargo/pull/14754)
+- Added a new build script invocation `cargo::error=MESSAGE` to report error messages.
+  ([docs](https://doc.rust-lang.org/nightly/cargo/reference/build-scripts.html#cargo-error))
+  [#14743](https://github.com/rust-lang/cargo/pull/14743)
+
 ### Changed
 
-- Pass `--no-tags` by default to git CLI when `net.git-fetch-with-cli = true`.
-  [#14688](https://github.com/rust-lang/cargo/pull/14688)
-- Several resolver performance enhancements, including
+- ❗️ cargo-publish: Always include Cargo.lock in published crates.
+  Originally it was only included for packages that have executables or examples
+  for use with `cargo install`.
+  [#14815](https://github.com/rust-lang/cargo/pull/14815)
+- Several resolver performance enhancements, including:
   - Use `rustc-hash` to speed up resolver
     [#14663](https://github.com/rust-lang/cargo/pull/14663)
   - Share conflict cache between activation retries
     [#14692](https://github.com/rust-lang/cargo/pull/14692)
   - Avoid cloning when iterating using `RcVecIter`
     [#14690](https://github.com/rust-lang/cargo/pull/14690)
+  - Avoid `HttpRegistry` returning `block_until_ready` too early when work is still pending
+    [#14694](https://github.com/rust-lang/cargo/pull/14694)
```

Comment:
There a better way to frame this for users?

Reply:
I don't think normal users care about details, so I went ahead removing details of these performance improvements.